### ISSUE TITLE
Results page: show unknown if DOB/age blank

### ISSUE THREAD

**Issue by nurifreembd** [FreeUKGen/FreeBMD2#604]
Results page: Age at Death/Date of Birth - currently shows blank cell. Suggestion is to show 'unknown' if no record of DOB/age.

**Comment by AlOneill** [comment]
I suggest that 'not recorded' (or similar) would be more accurate and honest than 'unknown' in most instances.

**Comment by nurifreembd** [comment]
sure, better then leaving it blank.

**Comment by KristinaGadzhieva** [comment]
Can we show 'Not applicable' for DOB, as well as for Mother's Maiden Name, Spouse's Surname? Blank cells in the table look unclear, either there is no data, or there is an error, or the table is not completed. @nurifreembd @AlOneill @richardofsussex

**Comment by AlOneill** [comment]
'Applicable' requires a higher reading age (in English) than 'recorded'. We should opt for the simplest way to express that the data was not collected at that time. 'Unknown' is bad because it is not true. Blank is bad because it looks as if something has gone wrong.

**Comment by nurifreembd** [comment]
Another option is to leave the cell blank and maybe above the table
consider a line something like ' we have most data on record but there may
be blank records where registration details were unrecorded at the time of
registration.

**Comment by KristinaGadzhieva** [comment]
How about 'No data' ? It seems quite simple and clear.

**Comment by nurifreembd** [comment]
works.

**Comment by AlOneill** [comment]
@KristinaGadzhieva Yes! Perfect.

**Comment by AlOneill** [comment]
@nurifreebmd Too long and complicated. Also, very likely to be missed by a screen reader user who goes straight to the table.

**Comment by richardofsussex** [comment]
@AlOneill presumably your comment relates to Nuri's suggestion? I was going to suggest 'No data' myself.

**Comment by AlOneill** [comment]
To clarify, I think 'No data' is (almost) perfect. (Almost because a user might ask, "and why not?" — my literal brain is on overdrive!! )

**Comment by richardofsussex** [comment]
I think that can be addressed in the help, where we can explain that the GRO have changed the information they record in the index over the years, and we can only offer the data actually recorded for any given event.

**Comment by KristinaGadzhieva** [comment]
Sure, we can provide clarification on the Help page. So I think we all agreed on 'No data'.

**Comment by nurifreembd** [comment]
My personal view is unknown works too as the charity is not responsible for
what is or not registered at the time of registration. we do not maintain
the data.

**Comment by richardofsussex** [comment]
OK, now implemented:
![image](https://github.com/user-attachments/assets/d2d7ffb8-0008-4507-b2c2-906209ab0f2f)
As you will see, I have also changed the case of the event type and made it singular. @Vino-S the changes are in fb_604_rbl, if you could please push them to test3.

**Comment by KristinaGadzhieva** [comment]
Thank you very much everyone!
@richardofsussex Richard, there are several empty cells in the screenshot. They don't have data either?

**Comment by richardofsussex** [comment]
Those are cells which are not relevant to that particular event type, and so can never contain data (e.g. Mother's Maiden Name and Age at Death for a marriage record). We _could_ put 'No data' in those as well, but personally I think it would make the listing look cluttered.

**Comment by KristinaGadzhieva** [comment]
Yes, it would be cluttered. We can leave as it is for now and wait for user feedback.

**Comment by nurifreembd** [comment]
All,
My personal view is we go back to drawing room. I do not agree with
inconsistent approach. One other option is for all unrecorded details is
show - and a line above the table somewhere stating - denotes no record
detail found or no record found or no data found

Or have NDF and somewhere above the table state NDF no data found

It looks cluttered and inconsistent with no data and other cells left blank

**Comment by KristinaGadzhieva** [comment]
We can show ND in each cell and make a note below the table for users explaining the meaning.

**Comment by richardofsussex** [comment]
Add dashes in currently blank cells.

**Comment by KristinaGadzhieva** [comment]
@AlOneill do you agree with the proposed approach: 
- to display a dash in cells which are not relevant to a particular event type 
- to show 'no data' in cells for which the data was not collected
and add a note below the table with explanation for users ?

**Comment by richardofsussex** [comment]
Implemented:
![image](https://github.com/user-attachments/assets/c175b9c9-c15b-4bd3-b5d2-76be37f7e9bc)
@Vino-S please push to test3.

**Comment by nurifreembd** [comment]
Looks good Richard. Looks presentable with the dash.

**Comment by richardofsussex** [comment]
Now on test3 - thanks @Vino-S. (BTW, I also changed e.g. 'BIRTHS' to 'Birth' as discussed. Bonus!)

**Comment by KristinaGadzhieva** [comment]
I think it would be worth adding an explanation of the difference between a dash and 'No data' below the table for users.

**Comment by nurifreembd** [comment]
Yes, I agree.

**Comment by AlOneill** [comment]
If you are going to explain anything about the table, please **add a heading** to that explanation so that a screen reader user can discover it (without having to read the whole page).

**Comment by KristinaGadzhieva** [comment]
@AlOneill @richardofsussex @nurifreembd 
Thank you. How about one of these headings:  Table details / Table interpretation / Data navigation /Table navigation / How to read the table? / How to read the data?

**Comment by richardofsussex** [comment]
I must say, that starts to sound like [part of] a Help page. Maybe that is where such detailed advice should live? Personally I quite like the FreeReg results page layout, which has a brief summary of what you looked for and what we found, followed by a table of results. (You'll notice there is also a brief summary there of how to sort the results table.)

**Comment by KristinaGadzhieva** [comment]
Commented in #609: I have added information about 'No data' and dash (‘-’) cases to the Help MVP doc.
A short note for users under the table about 'No data' and dash (‘-’) cases in story #609.

**Comment by KristinaGadzhieva** [comment]
@Vino-S This story ready for deployment.

**Comment by nurifreembd** [comment]
Kristina,

Ideal to add a note below the table

**Comment by richardofsussex** [comment]
Note, with accessibility heading, added. (I copied Kristina's help text.) @Vino-S please push code `fb_604_rbl_3` to test3.

**Comment by KristinaGadzhieva** [comment]
Thank you, Richard. @Vino-S could you please deploy it?

**Comment by Vino-S** [comment]
Deployed

**Comment by KristinaGadzhieva** [comment]
I don't see a note about dash cases under the table on Beta.

**Comment by richardofsussex** [comment]
It's in the branch `fb_rbl_604_3` (N.B. @Vino-S I mucked up the branch naming!)

**Comment by Vino-S** [comment]
Deployed
